src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -1191,21 +1138,59 @@ public ActivityIdFormat IdFormat
             private set => _state = (_state & ~State.FormatFlags) | (State)((byte)value & (byte)State.FormatFlags);
         }
 
-        /// <summary>
-        /// Having our own key-value linked list allows us to be more efficient
-        /// </summary>
         private partial class LinkedListNode<T>
         {
             public LinkedListNode(T value) => Value = value;
-            public LinkedListNode(T value, LinkedListNode<T>? next)
-            {
-                Value = value;
-                Next = next;
-            }
             public T Value;
             public LinkedListNode<T>? Next;
         }
 
+        private class LinkedList<T>
+        {
+            private LinkedListNode<T> _first;
+            private LinkedListNode<T> _last;
+
+            public LinkedList(T firstValue) => _last =_first = new LinkedListNode<T>(firstValue);
+
+            public LinkedList(IEnumerable<T> list)
+            {
+                IEnumerator<T> enumerator = list.GetEnumerator();
+                Debug.Assert(enumerator.MoveNext());
```

Comment:
It won't matter if you change this based on my previous comment (e.g. deleting the above two lines), but this MoveNext call is going to be compiled out in a release build as written.

Reply:
Yes, I'll apply your suggestion.